src/actions/userActions.js

```diff
@@ -47,70 +52,65 @@ export function authenticateUser(userCreds) {
             return response.json();
         }).then(json => {
             if (!json.success) {
-                dispatch(errorUser(userCreds, json.message));
+                localStorage.removeItem('token');
+                dispatch(ActionUtil.error(500, json.message));
                 return;
             }
 
-            localStorage.setItem('userId', json.data.id);
-            localStorage.setItem('roasterId', json.data.roasterId);
+            dispatch(getUserInfo());
             dispatch(receiveUser(json));
         }).catch(err => {
-            dispatch(errorUser(userCreds, err.message));
+            localStorage.removeItem('token');
+            dispatch(ActionUtil.error(500, err.message));
         });
     };
 }
 
-export function getUserInfo(userId) {
+export function getUserInfo() {
     return dispatch => {
-        return fetch(USER_URL + userId, ActionUtil.auth({
+        return fetch(USER_URL, ActionUtil.auth({
             method: 'GET'
-        })).then((response) => {
+        })).then(response => {
             return response.json();
-        }).then((json) => {
-            if(!json.success) {
-                dispatch(errorUser(userId, json.message));
+        }).then(json => {
+            if (!json.success) {
+                dispatch(ActionUtil.error(500, json.message));
                 return;
             }
 
-            localStorage.setItem('roasterId', json.data.roasterId);
+            if (json.data.roasterId) {
+                dispatch(getRoaster(json.data.roasterId));
+            }
             dispatch(receiveUser(json));
-        }).catch((err) => {
-            dispatch(errorUser(userId, err));
-        })
-    }
+        }).catch(err => {
+            dispatch(ActionUtil.error(500, err.message));
+        });
+    };
 }
 
 export function updateUserInfo(userInfo, userId) {
-  return dispatch => {
-      return fetch(USER_URL + userId, ActionUtil.auth({
-          method: 'PUT',
-          body: JSON.stringify(userInfo)
-      })).then((response) => {
-          return response.json();
-      }).then((json) => {
-        if(!json.success) {
-          dispatch(errorUser(userInfo, json.message));
-          return;
-        }
-
-        dispatch(receiveUser(json))
-      }).catch((err) => {
-          dispatch(errorUser(userInfo, err));
-      });
-  };
-}
+    return dispatch => {
+        return fetch(USER_URL + userId, ActionUtil.auth({
+            method: 'PUT',
+            body: JSON.stringify(userInfo)
+        })).then(response => {
+            return response.json();
+        }).then(json => {
+            if (!json.success) {
+                dispatch(ActionUtil.error(500, json.message));
+                return;
+            }
 
-function errorUser(userInfo, err) {
```

Comment:
We should try and reserve custom error types only for when they cannot be handled by the dashboard component.